addons/website_sale/controllers/main.py

```diff
@@ -510,6 +510,7 @@ def cart(self, access_token=None, revive='', **post):
         access_token: Abandoned cart SO access token
         revive: Revival method when abandoned cart. Can be 'merge' or 'squash'
         """
+        request.website.sale_get_order(update_pricelist=True)
```

Comment:
Not sure we want to do it here or like that. This is also used to revive an old cart, where the comportment might be different. I don't know well what is expected in that case, might be worth it to make sure it's okay for that too, or only do it if it's not an abandoned cart.
cc: @paju-odoo

Reply:
I think it's okay to do it for abandoned cart as well. The pricelist might have changed in the meantime and if it still applies, it makes sense for the user to see the new price :) 
We just have to make sure that the conditions are met
cc @Demesmaeker